src/EventSourcing/SnapshottingEventSourcingRepository.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the broadway/broadway-snapshotting package.
+ *
+ * (c) Qandidate.com <opensource@qandidate.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Broadway\Snapshotting\EventSourcing;
+
+use Broadway\Domain\AggregateRoot;
+use Broadway\EventSourcing\EventSourcingRepository;
+use Broadway\EventStore\EventStore;
+use Broadway\Repository\Repository;
+use Broadway\Snapshotting\Snapshot\Snapshot;
+use Broadway\Snapshotting\Snapshot\SnapshotNotFoundException;
+use Broadway\Snapshotting\Snapshot\SnapshotRepository;
+use Broadway\Snapshotting\Snapshot\Trigger;
+
+class SnapshottingEventSourcingRepository implements Repository
+{
+    private $eventSourcingRepository;
+    private $eventStore;
+    private $snapshotRepository;
+    private $trigger;
+
+    public function __construct(
+        EventSourcingRepository $eventSourcingRepository,
+        EventStore $eventStore,
+        SnapshotRepository $snapshotRepository,
+        Trigger $trigger
+    ) {
+        $this->eventSourcingRepository = $eventSourcingRepository;
+        $this->eventStore              = $eventStore;
+        $this->snapshotRepository      = $snapshotRepository;
+        $this->trigger                 = $trigger;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function load($id)
+    {
+        try {
+            $snapshot = $this->snapshotRepository->load($id);
```

Comment:
Like said on this other PR, I'd not use an exception to tell the repository that snapshot is not available.
Exceptions are expensive (determine the stack trace), hence they should only be used for real errors.
A not existing snapshot for an aggregate is not an error. Someone cleaned the cache, snapshot trigger did not match yet ...

Just let the snapshotRepository return `NULL`

Question: The `$id` is globally unique? Otherwise an aggregate type is needed so that the snapshotRepository knows  which snapshot should be looked up